src/SourceControl/AzureDevOps/AzureDevOpsSourceControlAccessor.cs

```diff
@@ -0,0 +1,148 @@
+                            if (item.path.EndsWith("readme.md", StringComparison.CurrentCultureIgnoreCase))
+                            {
+                                readmeItems.Add(item);
+                            }
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                //TODO: handle exceptions
+                Console.WriteLine(ex);
+            }
+            return readmeItems;
+        }
+
+
+        private async Task<IEnumerable<Document>> GetReadmesFromItems(IEnumerable<Item> readmeItems, Guid repositoryId, AzureDevOpsMetadata metadata)
+        {
+            List<Task<Document>> readmeTasks = new List<Task<Document>>();
+
+            foreach (var item in readmeItems)
+            {
+                readmeTasks.Add(GetReadmeDocumentFromURL(item, repositoryId, metadata));
+            }
+
+            await Task.WhenAll(readmeTasks.ToArray());
+
+            List<Document> readmeDocuments = new List<Document>();
+            foreach (var task in readmeTasks)
+            {
+                readmeDocuments.Add(task.Result);
+            }
+            return readmeDocuments;
+        }
+
+        private async Task<Document> GetReadmeDocumentFromURL(Item item, Guid repositoryId, AzureDevOpsMetadata metadata)
+        {
+            try
+            {
+                using (HttpClient client = new HttpClient())
+                {
+                    client.DefaultRequestHeaders.Accept.Add(
+                        new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/octet-stream"));
+
+                    client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic",
+                        Convert.ToBase64String(
+                            System.Text.ASCIIEncoding.ASCII.GetBytes($":{metadata.PersonalAccessToken}")));
+
+                    using (var str = await client.GetStreamAsync(item.url))
+                    {
+                        using (StreamReader reader = new StreamReader(str))
+                        {
+                            var readmeContent = reader.ReadToEnd();
+                            return new Document
+                            {
+                                Body = readmeContent,
+                                RepositoryId = repositoryId,
+                                Path = item.path,
+                                //TODO: Get these from AzureDevOps API response
+                                Created = DateTime.UtcNow,
+                                Updated = DateTime.UtcNow
+                            };
+
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                //TODO: handle exceptions
+                Console.WriteLine(ex);
+                return null;
+            }
+        }
+
+        public class ItemsResponse
```

Comment:
might as well put these in their own files since this integration is already in its own folder

Reply:
done